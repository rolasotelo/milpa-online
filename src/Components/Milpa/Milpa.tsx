import React from "react";
import CacaoBoard from "../CacaoBoard/CacaoBoard";
import Crop from "../Crop/Crop";
import StatusBoard from "../StatusBoard/StatusBoard";

interface Props {}

const miMilpa = [
  { name: "🥕" },
  { name: "🌽" },
  { name: "🥕" },
  { name: "🍅" },
  { name: "🍅" },
  { name: "🫑" },
  { name: "🥕" },
  { name: "🌽" },
  { name: "🥔" },
  { name: "🥕" },
  { name: "🪰" },
  { name: "🌽" },
  { name: "🌶" },
  { name: "🥔" },
  { name: "🍅" },
  { name: "🌽" },
];

const Milpa = (props: Props) => {
  return (
    <div className="flex flex-col bg-mexicanGreen-light w-1/3">
      <div className="py-4">
        <StatusBoard />
      </div>
      <div className="m-4 p-6 bg-yellow-800 grid grid-cols-4 text-center gap-4">
        {miMilpa.map((crop, index) => {
          return <Crop key={index} text={crop.name} />;
        })}
      </div>
      <div className="py-4">
        <CacaoBoard />
      </div>
    </div>
  );
};

export default Milpa;
