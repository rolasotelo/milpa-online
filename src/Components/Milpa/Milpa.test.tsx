import React from "react";
import renderer from "react-test-renderer";

import Milpa from "./Milpa";

describe("<Milpa />", () => {
  const defaultProps = {
    isYourMilpa: true,
    milpa: [undefined],
    edges: {
      top: [undefined],
      bottom: [undefined],
      left: [undefined],
      right: [undefined],
    },
  };
  const wrapper = renderer.create(<Milpa {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
