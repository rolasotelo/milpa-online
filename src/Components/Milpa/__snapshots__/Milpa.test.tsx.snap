// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Milpa /> render 1`] = `
<div
  className="flex flex-col bg-mexicanGreen-light w-1/3"
>
  <div
    className="py-4"
  >
    <div
      className="h-16 bg-mexicanBoneLight"
    >
      Your Turn
    </div>
  </div>
  <div
    className="m-4 p-6 bg-yellow-800 grid grid-cols-4 text-center gap-4"
  >
    <div
      className="w-16 h-16 bg-yellow-900"
    >
      🥕
    </div>
    <div
      className="w-16 h-16 bg-yellow-900"
    >
      🌽
    </div>
    <div
      className="w-16 h-16 bg-yellow-900"
    >
      🥕
    </div>
    <div
      className="w-16 h-16 bg-yellow-900"
    >
      🍅
    </div>
    <div
      className="w-16 h-16 bg-yellow-900"
    >
      🍅
    </div>
    <div
      className="w-16 h-16 bg-yellow-900"
    >
      🫑
    </div>
    <div
      className="w-16 h-16 bg-yellow-900"
    >
      🥕
    </div>
    <div
      className="w-16 h-16 bg-yellow-900"
    >
      🌽
    </div>
    <div
      className="w-16 h-16 bg-yellow-900"
    >
      🥔
    </div>
    <div
      className="w-16 h-16 bg-yellow-900"
    >
      🥕
    </div>
    <div
      className="w-16 h-16 bg-yellow-900"
    >
      🪰
    </div>
    <div
      className="w-16 h-16 bg-yellow-900"
    >
      🌽
    </div>
    <div
      className="w-16 h-16 bg-yellow-900"
    >
      🌶
    </div>
    <div
      className="w-16 h-16 bg-yellow-900"
    >
      🥔
    </div>
    <div
      className="w-16 h-16 bg-yellow-900"
    >
      🍅
    </div>
    <div
      className="w-16 h-16 bg-yellow-900"
    >
      🌽
    </div>
  </div>
  <div
    className="py-4"
  >
    <div
      className="h-16 bg-mexicanBoneLight"
    >
      100 cacao
    </div>
  </div>
</div>
`;
